File: library_C++/数学/幾何/座標の回転.cpp
"""
ラジアン = 度 * (PI / 180)
cout << fixed << setprecision(10);
d = d * (PI / 180);
auto p = rotate(a, b, d);

"""


"""
原点周りで反時計回りに radian 回転

"""

pair<ld, ld> rotate(ld x, ld y, ld radian){
    ld nx = x * cos(radian) - y * sin(radian);
    ld ny = x * sin(radian) + y * cos(radian);
    return mp(nx, ny);
}

File: library_python/数学/幾何/座標の回転.py
# 座標の回転
import math
x1 = x * math.cos(theta) - y * math.sin(theta)
y1 = x * math.sin(theta) + y * math.cos(theta)

# 回転したい点が原点じゃないとき
# a, b 原点にしたい
a = (xx + x) / 2
b = (yy + y) / 2
x -= a
y -= b
x1 = x * math.cos(theta) - y * math.sin(theta)
y1 = x * math.sin(theta) + y * math.cos(theta)
x1 += a
y1 += b